cumulus/ansible/tasks/providers/ec2.py

```diff
@@ -0,0 +1,183 @@
+        cluster_id = str(cluster["_id"])
+
+        instances = list(self.ec2.instances.filter(Filters=[
+            {'Name': 'tag:ec2_pod_instance_name', 'Values': ['head']},
+            {'Name': 'tag:ec2_pod', 'Values': [cluster_id]},
+            {'Name': 'instance-state-name', 'Values': ['running']}]))
+
+        if len(instances) == 0:
+            raise Exception("No master node could be found!")
+        if len(instances) > 1:
+            raise Exception("More than one master node was found!")
+
+        return self._get_instance_vars(instances[0])
+
+
+    def get_volumes(self):
+        pass
+
+    @property
+    def ec2(self):
+        try:
+            return boto3.resource(
+                'ec2',
+                self.regionName,
+                aws_access_key_id=self.accessKeyId,
+                aws_secret_access_key=self.secretAccessKey)
+        except AttributeError:
+            return boto3.resource(
+                'ec2',
+                os.environ['REGION_NAME'],
+                aws_access_key_id=self.accessKeyId,
+                aws_secret_access_key=self.secretAccessKey)
+
+    def _get_volume(self, girder_volume, refresh_cache=False):
+        volume_id = parse('ec2.id').find(girder_volume)[0].value
+
+        try:
+            if refresh_cache:
+                self._volume_cache[volume_id] = self.ec2.Volume(volume_id)
+
+            return self._volume_cache[volume_id]
+        except KeyError:
+            self._volume_cache[volume_id] = self.ec2.Volume(volume_id)
+
+        return self._volume_cache[volume_id]
+
+    def get_volume(self, girder_volume):
+        aws_volume = self._get_volume(girder_volume)
+        volume = {'volume_id': aws_volume.id}
+
+        status = parse('VolumeStatuses[0].VolumeStatus.Status')\
+            .find(aws_volume.describe_status())[0].value
+
+        if status != 'ok':
+            if status == 'insufficient-data':
+                volume['state'] = VolumeState.PROVISIONING
+            else:
+                volume['state'] = VolumeState.ERROR
+
+        # TODO add 'instance_id' here (None if not attached)
+        if aws_volume.attachments:
+            volume['state'] =  VolumeState.INUSE
+        else:
+            volume['state'] = VolumeState.AVAILABLE
+
+        return volume
+
+Provider.register('ec2', EC2Provider)
+
+
+if __name__ == "__main__":
+    REQUIRED_ENV_VARS = ('AWS_ACCESS_KEY_ID', 'AWS_SECRET_ACCESS_KEY',
+                         'CLUSTER_ID', 'REGION_NAME')
+
+    for required_env_var in REQUIRED_ENV_VARS:
+        if os.environ.get(required_env_var, '') == '':
+            raise Exception('Required env var %s not given to Cumulus'
+                            'inventory.' % required_env_var)
+
+    p = Provider({
+        "accessKeyId": os.environ.get("AWS_ACCESS_KEY_ID"),
+        "secretAccessKey": os.environ.get("AWS_SECRET_ACCESS_KEY"),
+        "type": "ec2"
+    })
+
+    print(json.dumps(p.get_inventory(os.environ.get('CLUSTER_ID'))))
```

Comment:
This is part of a plan to do away with dynamic_inventory/ec2.py and use providers/ec2.py for all things related to pulling state from AWS. dynamic_inventory/ec2.py was originally designed to be run like ```ansible-playbook -i ec2.py foo.yml```  this ensures that providers/ec2.py replicates that functionality